web/src/app/(api)/shorten/route.ts

```diff
@@ -0,0 +1,61 @@
+import { URLShortenerInputField, urlShortenerSchema } from "@/core/schema/url";
+import { newPrefixedId } from "@/lib/id";
+import { connectDB } from "@/server/config/database";
+import { UrlStatus } from "@/server/models/url";
+import { urlRepo } from "@/server/repository/url.repo";
+import {
+  newBadRequestApiResponse,
+  newSuccessApiResponse,
+} from "@/server/req-res";
+import { customAlphabet } from "nanoid";
+import { NextRequest } from "next/server";
+
+export async function POST(req: NextRequest) {
+  await connectDB();
+  const body = (await req.json()) as URLShortenerInputField;
+
+  const { data, error } = urlShortenerSchema.safeParse(body);
+
+  if (error || data) {
+    return newBadRequestApiResponse({
+      message: "Validation Failed",
+      data: null,
+    });
+  }
+
+  const { original, userId } = data;
+
+  const existingURL = await urlRepo.findByOriginalAndUser(original, userId);
+
+  if (existingURL) {
+    return newSuccessApiResponse({
+      message: "URL already exist",
+      data: existingURL,
+    });
+  }
+
+  const nanoid = customAlphabet("1234567abcd", 6);
+
+  const shortId = nanoid();
+
+  const newURL = await urlRepo.create({
+    id: newPrefixedId("url"),
+    original,
+    shortId,
+    userId,
+    clicks: 0,
+    status: UrlStatus.active,
+  });
```

Comment:
First of all, move this `nanoid` implementation to the [id file in /lib](https://github.com/Vanessa082/minxy/blob/19c64ebd1308c74f6b660027982969f731260754/web/src/core/lib/id.ts) directory



```suggestion
  const newURL = await urlRepo.create({
    id: newPrefixedId("url"),
    original,
    shortId: newShortId(),
    userId,
    clicks: 0,
    status: UrlStatus.active,
  });
```